src/hotspot/share/ci/ciMethodData.cpp

```diff
@@ -68,7 +68,11 @@ class PrepareExtraDataClosure : public CleanExtraDataClosure {
   { }
 
   bool is_live(Method* m) {
-    if (!m->method_holder()->is_loader_alive()) {
+    Klass* holder = m->method_holder();
```

Comment:
Changes in this file seems not related and can be pushed/tested separately. If they are related - there should be condition for additional checks.

Reply:
You mean you want these checks to be done only if `TrainingData::have_data() == true` ?

Reply:
Yes, if it is related. Otherwise you may change default behavior when Leyden code is not used.

Reply:
I simplified it and factored out the new semantics.